internal/helper.go

```diff
@@ -0,0 +1,27 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package internal
+
+import "unsafe"
+
+// ByteSlice2String converts a byte slice into a string without a heap allocation.
+// Be aware that the byte slice and the string share the same memory - which makes
+// the string mutable.
+func ByteSlice2String(b []byte) string {
```

Comment:
Maybe we should call this `UnsafeByteSlice2String` so that it's very clear that there are special considerations.

Reply:
Conventionally this is named something along the lines of `unsafeString` (usually locally defined to avoid ignorance due to distance).

Reply:
Not really. In the standard library where unsafe string conversion is done it is either done directly, or in one case with a local function called that. In other code bases it is variously called that or some other things like `BytesToString` and so on. It does however fit with the Go approach to naming; it is short, without fluff, and says exactly what it is and has the nice property that it looks like a conversion.